### PR TITLE
KOGITO-1436: Remove double setContent call

### PR DESCRIPTION
**Pull request by ederign**
This PR changes set content method to return a Promise<Void> instead of void.

The goal of this change is to be able to make the loading of editors predictable and remove the need for double set content call.
Merge together with:
https://github.com/kiegroup/appformer/pull/934
https://github.com/kiegroup/kie-wb-common/pull/3246

### CODE REVIEW

**Comment by manstis** [comment]
@ederign IDK if you have another PR coming but you probably need to update [appformer](https://github.com/kiegroup/appformer/blob/master/uberfire-workbench/uberfire-workbench-processors/src/main/java/org/uberfire/annotations/processors/GeneratorUtils.java#L128) too?

**Comment by ederign** [comment]
@manstis absolutely, I'm creating right now  one  over https://github.com/kiegroup/kie-wb-common/pull/3236/files

File: packages/microeditor-envelope/src/EditorEnvelopeController.tsx [hunk 1]
```diff
@@ -58,9 +57,8 @@ export class EditorEnvelopeController {
         if (editor) {
           this.editorEnvelopeView!.setLoading();
           editor
-            .setContent("", "")
-            .finally(() => this.waitForEmptySetContentThenSetLoadingFinished())
-            .then(() => editor.setContent(contentPath, editorContent.content));
+            .setContent(contentPath, editorContent.content)
+            .finally(() => this.editorEnvelopeView!.setLoadingFinished());
```
Comment (tiagobento) [inline packages/microeditor-envelope/src/EditorEnvelopeController.tsx:61]: 🙏😱☝️👌😉

File: packages/microeditor-envelope/src/__tests__/EditorEnvelopeController.test.tsx [hunk 2]
```diff
@@ -21,6 +21,7 @@ import { EnvelopeBusMessage, EnvelopeBusMessageType } from "@kogito-tooling/micr
 import { LanguageData } from "@kogito-tooling/core-api";
 import { DummyEditor } from "./DummyEditor";
 import { ResourceContentEditorCoordinator } from "../api/resourceContent";
+import { promises } from "fs";
```
Comment (tiagobento) [inline packages/microeditor-envelope/src/__tests__/EditorEnvelopeController.test.tsx]: Please remove this line.

**Review by tiagobento — CHANGES_REQUESTED** [review]
Looks great. Just an extra import there. Please remove it. Thanks @ederign!

**Review by romartin — APPROVED** [review]

**Review by manstis — APPROVED** [review]

**Comment by jomarko** [comment]
### Manually tried scenarios
In VS code open
- [x] Valid bpmn, do changes, reopen
- [ ] Invalid bpmn, fix, reopen
- [x] Valid dmn, do changes, reopen
- [x] Invalid dmn, fix, reopen

**Comment by jomarko** [comment]
When user tries to open invalid BPMN process, ace/xml editor is not shown. Even if I confirm no such element popup from screenshot, ace/xml editor is not shown. Fr dmn it works fine.
Screenshot from 2020-03-26 15-41-24

File: packages/microeditor-envelope/src/__tests__/EditorEnvelopeController.test.tsx (hunk 2 above)
Comment (ederign) [reply packages/microeditor-envelope/src/__tests__/EditorEnvelopeController.test.tsx]: fixed.

**Comment by ederign** [comment]
@jomarko I've updated all the PR's as you requested.

I'm not sure if the invalid BPMN issue is related to changes on this PR. @romartin any idea?

UPDATED:
on @jomarko's comment on kie-wb-common
> Manual check in business central was same as on vs code [kiegroup/kogito-tooling#95 (comment)](https://github.com/kiegroup/kogito-tooling/pull/95#issuecomment-604468044), but passed all scenarios.

 I just notice that the same behaviour appears on BC, so probably it's unrelated because  afaik my changes, Roger and Michael was only on kogito. Probably this is a regression but I prefer to wait Roger to confirm.

**Review by tiagobento — APPROVED** [review]

**Comment by romartin** [comment]
@ederign @jomarko good catch about the invalid bpmn file guys. TBH no idea, but I don't think it should be related to this. 

I would suggest merging this, which is preventing several things to move forward, and report the "invalid bpmn issue" so I'll take a look later on. WDYT?

Thanks!!

**Comment by lazarotti** [comment]
@ederign @jomarko @romartin as this has blocked many other PRs I strongly suggest go ahead with the merge and create a new Jira focused only on invalid xml as it is in fact part of a different use case

**Comment by manstis** [comment]
@romartin there's an old JIRA for BPMN kicking around somewhere about falling back to ACE for invalid files. It is unrelated to this PR.

**Comment by jomarko** [comment]
@manstis if you have a link, would be good paste it here for a record.

**Comment by manstis** [comment]
@jomarko here's a link that just popped up in my inbox https://issues.redhat.com/browse/KOGITO-513.. I am sure there are others similar.

**Comment by jomarko** [comment]
@manstis thank you, I came across this one too.

**Pull request merged**